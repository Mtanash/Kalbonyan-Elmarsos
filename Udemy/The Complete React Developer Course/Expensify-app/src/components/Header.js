import React from "react";
import { Link } from "react-router-dom";
import { connect } from "react-redux";
import { startSignout } from "../actions/auth";

export const Header = ({ startSignout }) => (
  <header className="header">
    <div className="content-container">
      <div className="header__contnet">
        <Link className="header__title" to="/dashboard">
          <h1>Expensify</h1>
        </Link>
        <button className="button button--transparent" onClick={startSignout}>
          Logout
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startSignout: () => dispatch(startSignout()),
});

export default connect(undefined, mapDispatchToProps)(Header);
