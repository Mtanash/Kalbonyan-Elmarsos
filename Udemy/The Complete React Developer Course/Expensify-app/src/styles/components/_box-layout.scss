.box-layout {
  background: url(/images/bg.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  min-height: 100vh;
  min-width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
}

.box-layout__box {
  background: fade-out($color: #fff, $amount: 0.15);
  border-radius: 3px;
  text-align: center;
  width: 28rem;
  padding: $l-size $m-size;
}

.box-layout__title {
  margin: 0 0 $m-size 0;
  line-height: 1;
}
