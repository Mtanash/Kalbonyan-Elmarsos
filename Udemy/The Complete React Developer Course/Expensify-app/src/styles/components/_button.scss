.button {
  background-color: $blue;
  color: #fff;
  font-size: $font-size-large;
  font-weight: 500;
  padding: $s-size;
  outline: none;
  border: none;
  display: inline-block;
  text-decoration: none;
  cursor: pointer;
  line-height: 1;
}

.button--transparent {
  background-color: transparent;
}

.button--secondary {
  background-color: #888;
}
