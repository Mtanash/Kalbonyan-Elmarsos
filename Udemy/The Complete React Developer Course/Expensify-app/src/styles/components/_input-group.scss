.input-group {
  display: flex;
  gap: $s-size;
  flex-direction: column;
  align-items: center;
  margin-bottom: $s-size;

  @media (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin-bottom: $l-size;
  }
}
