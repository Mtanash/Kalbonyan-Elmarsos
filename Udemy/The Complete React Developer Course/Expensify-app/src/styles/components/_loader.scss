.loader {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  min-width: 100vw;
}

.loader__image {
  height: 6rem;
  width: 6rem;
}
