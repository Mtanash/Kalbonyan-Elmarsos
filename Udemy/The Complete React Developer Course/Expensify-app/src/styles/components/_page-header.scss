.page-header {
  background: $off-white;
  padding: $l-size;
  margin-bottom: $l-size;
}

.page-header__actions {
  margin-top: $l-size;
}

.page-header__title {
  margin: 0;
  font-weight: 300;
  span {
    font-weight: bold;
  }
}
