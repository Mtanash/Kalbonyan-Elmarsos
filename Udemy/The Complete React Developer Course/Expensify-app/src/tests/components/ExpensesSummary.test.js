import { ExpensesSummary } from "../../components/ExpensesSummary";
import React from "react";
import { shallow } from "enzyme";

test("should render ExpensesSummary with one expense", () => {
  const expensesCount = 1;
  const expensesTotal = 94.34;
  const wrapper = shallow(
    <ExpensesSummary
      expensesCount={expensesCount}
      expensesTotal={expensesTotal}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpensesSummary with two expense", () => {
  const expensesCount = 2;
  const expensesTotal = 94.34;
  const wrapper = shallow(
    <ExpensesSummary
      expensesCount={expensesCount}
      expensesTotal={expensesTotal}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
