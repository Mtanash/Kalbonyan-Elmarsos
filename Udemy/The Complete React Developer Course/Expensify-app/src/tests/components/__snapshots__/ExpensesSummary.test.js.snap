// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
         
        $0.94
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render ExpensesSummary with two expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span />
       
      expenses
       totalling
       
      <span>
         
        $0.94
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
